### PR TITLE
Use react-dropzone to improve file uploads

### PR DESCRIPTION
**Pull request by bhollis**
This improves our file uploads by using react-dropzone - we now have a single button that prompts for choosing a file and uploads it in one go, OR supports drag and drop of files onto the button. I also added extra validation and feedback for the DIM data import process and wish list import.

screen shot 2019-01-03 at 10 22 53 pm
screen shot 2019-01-03 at 10 22 59 pm

### CODE REVIEW

File: src/app/shell/rating-mode/RatingMode.tsx [hunk 1]
```diff
@@ -195,29 +191,37 @@ class RatingMode extends React.Component<Props, State> {
...
+          const inventoryCuratedRolls = dimCuratedRollService.getInventoryCuratedRolls(storeRolls);
+
+          const curationActionData = {
+            curationEnabled: dimCuratedRollService.curationEnabled,
+            inventoryCuratedRolls
+          };
+
+          store.dispatch(updateCurations(curationActionData));
+          refresh();
+          console.log({ inventoryCuratedRolls });
```
Comment (48klocs) [inline src/app/shell/rating-mode/RatingMode.tsx]: Probably don't need this debugging stub.

**Comment by 48klocs** [comment]
Code looks good, and getting rid of needing to click "find" after loading a file is a nice QOL improvement.

That said, the functionality seems a bit... subtle? My two concerns are that the buttons are small landing spots and that I'm used to seeing something with a dotted outline around it and an icon+text telling me to "upload files here". I know adding a second, kind of redundant, element would be tough on the popup because space is limited.

I don't think that should stop it from being merged, but piling a design check on to @inexorableAce for some point in the future would be good.

**Comment by inexorableAce** [comment]
yeah I think the typical recognized pattern for this kinda thing is the dotted border maybe something like this, doesn't take too much extra space either. 
image

**Comment by bhollis** [comment]
Makes sense, I can update it.

**Comment by bhollis** [comment]
screen shot 2019-01-05 at 12 13 35 am

How's this?

**Comment by inexorableAce** [comment]
Looks great 👍

**Comment by 48klocs** [comment]
Excellent!

**Pull request merged**